src/python/pants/jvm/goals/lockfile.py

```diff
@@ -20,56 +20,21 @@
 from pants.engine.rules import Get, collect_rules, rule
 from pants.engine.target import AllTargets
 from pants.engine.unions import UnionRule
-from pants.jvm.resolve import coursier_fetch, jvm_tool
+from pants.jvm.resolve import coursier_fetch
 from pants.jvm.resolve.common import ArtifactRequirement, ArtifactRequirements
 from pants.jvm.resolve.coursier_fetch import CoursierResolvedLockfile
-from pants.jvm.resolve.jvm_tool import GatherJvmCoordinatesRequest, JvmToolBase
 from pants.jvm.resolve.key import CoursierResolveKey
 from pants.jvm.resolve.lockfile_metadata import JVMLockfileMetadata
 from pants.jvm.subsystems import JvmSubsystem
 from pants.jvm.target_types import JvmArtifactCompatibleResolvesField
 from pants.util.logging import LogLevel
-from pants.util.meta import frozen_after_init
-from pants.util.ordered_set import FrozenOrderedSet
 
 
 @dataclass(frozen=True)
 class GenerateJvmLockfile(GenerateLockfile):
     artifacts: ArtifactRequirements
 
 
-@frozen_after_init
-@dataclass(unsafe_hash=True)
-class GenerateJvmLockfileFromTool:
-    artifact_inputs: FrozenOrderedSet[str]
-    options_scope: str
-    lockfile_dest: str
-
-    def __init__(self, tool: JvmToolBase) -> None:
-        # Note that `JvmToolBase` is not hashable, so we extract the relevant information eagerly.
-        self.artifact_inputs = FrozenOrderedSet(tool.artifact_inputs)
-        self.options_scope = tool.options_scope
-        self.lockfile_dest = tool.lockfile
```

Comment:
Moved to `jvm_tool.py` so that this file doesn't leak tool-specific code into it. Note that `jvm_tool.py` now has a dependency on `lockfile.py`, which was not the case before.

Reply:
Cool, agreed :) might be a confusing route to get there though!